pkg/cmd/auth/login/login_test.go

```diff
@@ -432,13 +438,21 @@ func Test_loginRun_Survey(t *testing.T) {
 			opts: &LoginOptions{
 				Interactive: true,
 			},
-			askStubs: func(as *prompt.AskStubber) {
-				as.StubPrompt("What account do you want to log into?").AnswerWith("GitHub Enterprise Server")
-				as.StubPrompt("GHE hostname:").AnswerWith("brad.vickers")
-				as.StubPrompt("What is your preferred protocol for Git operations?").AnswerWith("HTTPS")
-				as.StubPrompt("Authenticate Git with your GitHub credentials?").AnswerWith(false)
-				as.StubPrompt("How would you like to authenticate GitHub CLI?").AnswerWith("Paste an authentication token")
-				as.StubPrompt("Paste your authentication token:").AnswerWith("def456")
+			prompterStubs: func(pm *prompter.PrompterMock) {
+				pm.SelectFunc = func(prompt, _ string, _ []string) (int, error) {
+					switch prompt {
+					case "What account do you want to log into?":
+						return 1, nil
+					case "What is your preferred protocol for Git operations?":
+						return 0, nil
+					case "How would you like to authenticate GitHub CLI?":
+						return 1, nil
+					}
+					return -1, nil
```

Comment:
Instead of returning `-1` (which is an invalid result) on errors, we could return an error object describing the case that the test stub isn't sufficient